src/ipa_pc/mod.rs

```diff
@@ -1204,146 +1223,214 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         // to the internal state of the Fiat-Shamir
-        fs_rng.absorb(&to_bytes![batch_commitment].unwrap());
-        let point: G::ScalarField = fs_rng.squeeze_128_bits_challenge();
-
-        // Values: v_i = p_i(x), where x is fresh random challenge
-        let mut batch_values = vec![G::ScalarField::zero(); labeled_polynomials.len()];
-        for labeled_polynomial in labeled_polynomials.iter() {
-            batch_values[
-                *labels
-                    .get(labeled_polynomial.label())
-                    .ok_or(Error::MissingEvaluation {
-                        label: labeled_polynomial.label().to_string(),
-                    })?
-            ] = labeled_polynomial.polynomial().evaluate(point);
+        fs_rng.absorb(&to_bytes![h_comm].unwrap());
+        let x_point: G::ScalarField = fs_rng.squeeze_128_bits_challenge();
+
+        // LC(p_1(X),p_2(X),...,p_m(X),h(X))
+        let mut lc_polynomial = Polynomial::zero();
+        let mut lc_randomness = Randomness::empty(lc_segments_count);
+
+        let mut cur_challenge = G::ScalarField::one();
+
+        for (label, (_point_label, point)) in query_set.iter() {
+            let labeled_polynomial =
+                *poly_map.get(label).ok_or(Error::MissingPolynomial {
+                    label: label.to_string(),
+                })?;
+
+            let labeled_randomness =
+                *rands_map.get(label).ok_or(Error::MissingRandomness {
+                    label: label.to_string(),
+                })?;
+
+            // y_i
+            let y_i = labeled_polynomial.polynomial().evaluate(*point);
+
+            // (X - x_i)
+            let x_polynomial = Polynomial::from_coefficients_slice(&[
+                (G::ScalarField::zero() - point),
+                G::ScalarField::one()
+            ]);
+
+            // z_i(X) = z(X) / (X - x_i)
+            let z_i_polynomial = &z_polynomial / &x_polynomial;
+
+            // z_i(x)
+            let z_i_value = z_i_polynomial.evaluate(x_point);
+
+            // (p_i(X) - y_i)
+            let polynomial =
+                &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_slice(&[y_i]));
+
+            // LC(p_1(X),p_2(X),...,p_m(X)) = SUM ( lamda^i * z_i(x) * (p_i(X) - y_i) )
+            lc_polynomial +=
+                (cur_challenge, &(polynomial * &Polynomial::from_coefficients_slice(&[z_i_value])));
+
+            if has_hiding {
+                lc_randomness.rand = lc_randomness.rand.iter().enumerate().map(|(index, &lc_rand_segment)| {
+                    if labeled_randomness.randomness().rand.len() > index {
+                        lc_rand_segment +
+                            labeled_randomness.randomness().rand[index] * cur_challenge * z_i_value
+                    } else {
+                        lc_rand_segment
+                    }
+                }).collect::<Vec<_>>();
+            }
+
```

Comment:
not needed, as we do not commit it. (see below)

Reply:
Actually we still need this part of code, because each labeled_commitment can have hiding, and we need to combine it randomness to the lc_randomness